function.rb

```diff
@@ -0,0 +1,9 @@
+
+def function(n)
+  if n == 0
+    1
+  else
+    n * function(n-1)
+  end
```

Comment:
Does the new value of n need to be available outside of the function? It should be saved in a variable and returned.